cloudevents/extensions/opcua.md

```diff
@@ -0,0 +1,102 @@
+
+### source
+
+MUST either map to [Application Description](https://reference.opcfoundation.org/Core/Part4/v104/docs/7.1) field `applicationUri` of the OPC UA server or to an customer configured identifier like an unified namespace path.
+
+### type
+
+MUST map to [Data Set Message Header](https://reference.opcfoundation.org/Core/Part14/v105/docs/7.2.5.4#Table164) field `MessageType`.
+
+## Mapping of OPTIONAL Attributes
+
+### datacontenttype
+
+SHALL be `application/opcua+json` for OPC UA PubSub JSON payload and MAY be extended by `+gzip` when payload is gzip compressed.
+
+### dataschema
+
+OPC UA provides type information as part of PubSub metadata messages, for non OPC UA consumers or when different payload encoding like Avro is used, it is REQUIRED to provide schema information (based on metadata information) in a separate format like [JSON schema](https://json-schema.org/specification) or [Avro schema](https://avro.apache.org/docs/1.11.1/specification/) or others. For those cases the Attribute references the schema and is used for versioning.
+
+### subject
+
+For metadata and data messages (type one of `ua-metadata`, `ua-keyframe`, `ua-deltaframe`) MUST map to either [Data Set Message Header](https://reference.opcfoundation.org/Core/Part14/v105/docs/7.2.5.4#Table164) field `DataSetWriterId` or `DataSetWriterName`.
+
+For event messages (type equals to `ua-event`) SHALL map to [Base Event Type](https://reference.opcfoundation.org/Core/Part5/v104/docs/6.4.2) field `EventId`. 
+
+### time
+
+MUST map to [Data Set Message Header](https://reference.opcfoundation.org/Core/Part14/v105/docs/7.2.5.4#Table164) field `Timestamp`.
+
+## Mapping for other extensions
+
+The following well-known extensions attributes are used for data messages and event messages (type one of `ua-keyframe`, `ua-deltaframe`, `ua-event`).
+
+### sequence
+
+Attribute as defined by [sequence extensions](./sequence.md) MUST map to [Data Set Message Header](https://reference.opcfoundation.org/Core/Part14/v105/docs/7.2.5.4#Table164) field `SequenceNumber`.
+
+### traceparent
+
+Attribute as defined by [distributed-tracing extension](./distributed-tracing.md) SHALL be used to allow tracing from event publisher towards consumer.
+
+### tracestate
+
+Attribute as defined by [distributed-tracing extension](./distributed-tracing.md) might OPTIONAL be used to allow tracing from event publisher towards consumer.
+
+### recordedtime
+
+Attribute as defined by [recordedtime extension](./recordedtime.md) SHALL be used to determine the latency between event publisher towards consumer. 
+
+## Attributes
+
+### opcuametadatamajorversion
+
+- Type: `Integer`
+- Description: Links dataset message to the current version of the metadata. Contains value from `MajorVersion` of [Data Set Message Header](https://reference.opcfoundation.org/Core/Part14/v105/docs/7.2.5.4#Table164) field `MetaDataVersion`. 
+- Constraints
+  - OPTIONAL
+  - Can be omitted if `dataschema` is used
```

Comment:
s/Can/MAY/

Reply:
Actually, it's already OPTIONAL - so do we need this line at all or are you saying that it's require when `dataschema` not used?

Reply:
same for attributes below

Reply:
updated